vimoutliner/scripts/otl2html.py

```diff
@@ -1,216 +1,153 @@
-# Basic function
-#
-#    This program accepts text outline files and converts them
-#    to HTML.  The outline levels are indicated by tabs. A line with no
-#    tabs is assumed to be part of the highest outline level.
-#
-#    10 outline levels are supported.  These loosely correspond to the
-#    HTML H1 through H9 tags.  Alphabetic, numeric and bullet formats
-#    are also supported.
-#
-#    CSS support has been added.
-#
-
-###########################################################################
-# include whatever mdules we need
-
+import argparse
 import locale
 import os
 import re
 import sys
 import time
 
-###########################################################################
-# global variables
-
-formatMode = "indent"
-copyright = ""
-level = 0
-div = 0
-silentdiv = 0
-slides = 0
-hideComments = 0
-showTitle = 1
-inputFile = ""
 outline = []
 flatoutline = []
-inBodyText = 0        # 0: no, 1: text, 2: preformatted text, 3: table
-styleSheet = "nnnnnn.css"
-inlineStyle = 0
-
-###########################################################################
-# function definitions
-
-# usage
-# print the simplest form of help
-# input: none
-# output: simple command usage is printed on the console
-
-
-def showUsage():
-    print """
-    Usage:
-        otl2html.py [options] inputfile > outputfile
-    Options
-        -p              Presentation: slide show output for use with
-                        HtmlSlides.
-        -D              First-level is divisions (<div> </div>) for making
-                        pretty web pages.
-        -s sheet        Use the specified style sheet with a link. This is the
-                        default.
-        -S sheet        Include the specified style sheet in-line the
-                        output. For encapsulated style.
-        -T              The first line is not the title. Treat it as
-                        outline data
-        -c              comments (line with [ as the first non-whitespace
-                        character. Ending with ] is optional.
-        -C copyright    Override the internal copyright notice with the
-                        one supplied in the quoted string following this
-                        flag. Single or double quotes can be used.
-        -H              Show the file syntax help.
-    output is on STDOUT
-      Note: if neither -s or -S are specified, otl2html.py will default
-            to -s. It will try to use the css file 'nnnnnn.css' if it
-            exists. If it does not exist, it will be created automatically.
-    """
 
+class SyntaxAction(argparse.Action):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        if nargs is not None:
+            raise ValueError("nargs not allowed")
+        super(SyntaxAction, self).__init__(option_strings, dest, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        print '''
```

Comment:
This whole block could be made more readable (due to indentation) using:

    from textwrap import dedent
    print dedent(
        '''\
        Syntax
        Syntax is Vim Outliner's normal syntax. The following are supported:

            Text
        :    Body text marker. This text will wrap in the output.
        ;    Preformmated text. This text will will not wrap.
        ...
        ''')

For another example, see also: https://twitter.com/raymondh/status/945000654547775489

Reply:
I'm not a maintainer of `vimoutliner`. It's not in my power to merge it ;)

Reply:
I didn’t say that. Just put up your branch in your fork of this repository (``exhum/vimoutliner``). You can then set up pull request to ``mcepl/vimoutliner@argparse``, or you can just announce it here and I will do the rest.

Reply:
I don't understand what you want me to do. This is *your* pull-request. Not mine. I just added some feedback on it after you asked me to do a review. I closed #149 now as it seems to cause confusion.

Reply:
Yes, and you can add your pull-request on the top of mine another pull-request. Whatever, I will do it myself.

Reply:
> [...] you can add your pull-request on the top of mine another pull-request

There is no point doing that. The changes are almost identical. The only differences existed because I did my best to make *only* `argparse` changes in the PR. Your commit also mixes in PEP8 changes and replaces global vars by the argparse Namespace.

Adding my PR to the chain will not add anything useful. It can be ignored.